### PR TITLE
handle all Throwable instances during serialization

### LINKED ISSUE #2095: Not working Exception Handling in ExceptionController with PHP 7
Please fix the bug:
php.CRITICAL: Uncaught Error: Argument 3 passed to FOS\RestBundle\Controller\ExceptionController::getTemplateData() must be an instance of Exception, instance of TypeError given, called in /var/www/html/vendor/friendsofsymfony/rest-bundle/Controller/ExceptionController.php on line 74
It´s already reported in https://github.com/FriendsOfSymfony/FOSRestBundle/issues/2092 after it was closed.

### PR DESCRIPTION
**Pull request by xabbuh**
fixes #2095, replaces #2104

### CODE REVIEW

File: Serializer/Normalizer/AbstractExceptionNormalizer.php [hunk 1]
```diff
@@ -44,14 +44,14 @@ public function __construct(ExceptionValueMap $messagesMap, $debug)
...
      * Extracts the exception message.
      *
-     * @param \Exception $exception
+     * @param \Throwable $exception
      * @param int|null   $statusCode
      *
      * @return string
      */
-    protected function getExceptionMessage(\Exception $exception, $statusCode = null)
+    protected function getExceptionMessage(\Throwable $exception, $statusCode = null)
```
Comment (GuigZ-) [inline Serializer/Normalizer/AbstractExceptionNormalizer.php]: You change a definition of protected method, if someone overload this method, the code will be uncompatible.

Comment (xabbuh) [reply Serializer/Normalizer/AbstractExceptionNormalizer.php]: Thank you for the heads up. In fact, here this change is safe as the `AbstractExceptionNormalizer` class is marked as `@internal` which means that we do not need to aim for BC here.

File: Serializer/Normalizer/AbstractExceptionNormalizer.php [hunk 2]
```diff
@@ -44,17 +44,17 @@ public function __construct(ExceptionValueMap $messagesMap, $debug)
...
+     * @param \Throwable $throwable
      * @param int|null   $statusCode
      *
      * @return string
      */
-    protected function getExceptionMessage(\Exception $exception, $statusCode = null)
+    protected function getMessageFromThrowable(\Throwable $throwable, $statusCode = null)
     {
-        $showMessage = $this->messagesMap->resolveException($exception);
+        $showMessage = $this->messagesMap->resolveThrowable($throwable);
```
Comment (yceruto) [inline Serializer/Normalizer/AbstractExceptionNormalizer.php]: I can't find the `resolveThrowable()` method here:
https://github.com/FriendsOfSymfony/FOSRestBundle/blob/master/Util/ExceptionValueMap.php

How it should work?

Comment (xabbuh) [reply Serializer/Normalizer/AbstractExceptionNormalizer.php]: It's only present in 2.x: https://github.com/FriendsOfSymfony/FOSRestBundle/blob/2.7/Util/ExceptionValueMap.php

The code for 3.0 was fixed in #2112.

**Pull request merged**